### ISSUE TITLE
Warning host/device function call

### ISSUE THREAD

**Issue by PrometheusPi** [ComputationalRadiationPhysics/picongpu#848]
On taurus/zih I get the following warning:

```
Warning: calling a __host__ function("std::_List_base<picongpu::ISimulationPlugin *, std::allocator<picongpu::ISimulationPlugin *> > ::~_List_base")
  from a __host__ __device__ function("std::_List_base<picongpu::ISimulationPlugin *, std::allocator<picongpu::ISimulationPlugin *> > ::~_List_base [subobject]")
  is not allowed
```

I am using:
- `gcc/4.6.2`
- `boost/1.54.0`
- `cuda/5.0.35`

**Comment by psychocoderHPC** [comment]
Is this the full error message? If not please post all lines.

**Comment by PrometheusPi** [comment]
This is the entire message.

**Comment by psychocoderHPC** [comment]
I saw this warning now also with cuda8.

Loaded modules:
  1) gcc/4.9.2                     4) openmpi/1.8.6.kepler.cuda80   7) numactl/2.0.7
  2) cmake/3.3.0                   5) boost/1.56.0                  8) valgrind/3.8.1
  3) cuda/8.0                      6) hdf5-parallel/1.8.15

**Comment by psychocoderHPC** [comment]
solved with #1593
